src/main/java/lotto/domain/model/LottoDispenser.java

```diff
@@ -0,0 +1,46 @@
+package lotto.domain.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import lotto.constant.IllegalStateExceptionType;
+import lotto.domain.LottoGenerateStrategy;
+
+public final class LottoDispenser {
+    private final LottoGenerateStrategy lottoGenerateStrategy;
+    private final LottoPurchaseCost lottoPurchaseCost;
+    private boolean isDispensed;
+
+    public LottoDispenser(LottoGenerateStrategy lottoGenerateStrategy, LottoPurchaseCost lottoPurchaseCost) {
+        this.lottoGenerateStrategy = lottoGenerateStrategy;
+        this.lottoPurchaseCost = lottoPurchaseCost;
+        isDispensed = false;
+    }
+
+    private void validateIsDispensed() {
+        if (isDispensed) {
+            throw IllegalStateExceptionType.LOTTO_ALREADY_DISPENSED.getException();
+        }
+    }
+
+    private Lotto generateLotto() {
+        return lottoGenerateStrategy.generate();
+    }
+
+    private List<Lotto> generateLottos(int lottoAmount) {
+        List<Lotto> elements = new ArrayList<>();
+
+        for (int i = 0; i < lottoAmount; i++) {
+            elements.add(generateLotto());
+        }
+
+        return elements;
```

Comment:
```suggestion
        return IntStream.range(0, lottoAmount)
                .mapToObj(this::generateLotto)
                .toList()
```
스트림으로 더 간결하게 줄여볼 수 있을 것 같아요!

Reply:
맞습니다! 가독성이 확실히 더 좋은 것 같습니다 👍🏼